### PR TITLE
Provider logo, sorting providers, and showing localized Connect UI page

### PR DESCRIPTION
**Pull request by petalvlad**
Hello guys. 

I've made some changes in the lib and example project, hope they will be helpful.

- Added `logoURL` attribute into `SEProvider` model
- Displayed logos in providers list (I had to add `SVGKit` and `SDWebImageSVGCoder` pods because logos are in SVG format)
- Sorted providers by country, and placed providers from current country at the top of the list
- Included `locale` into `SEAttempt` of `SECreateSessionsParams` payload, so that the Connect UI page returned localized

### CODE REVIEW

File: Example/saltedge-ios/View Controllers/ProvidersViewController.swift [hunk 1]
```diff
@@ -27,25 +27,55 @@ class ProvidersViewController: UIViewController {
...
+                self.filteredProviders = providers
+                self.tableView.reloadData()
             case .failure(let error):
                 HUD.flash(.labeledError(title: "Error", subtitle: error.localizedDescription), delay: 3.0)
             }
         }
     }
+    
+    private func sortProvidersByCountry(_ providers: [SEProvider], startingWith topCountry: String? = nil) -> [SEProvider] {
+        return providers.sorted(by: {
```
Comment (baller784) [inline Example/saltedge-ios/View Controllers/ProvidersViewController.swift:61]: Looks good, but replace with a better code style, like this:

```swift
return providers.sorted(
        by: {
            if $0.countryCode != $1.countryCode {
                if $0.countryCode == topCountry || $1.countryCode == topCountry {
                    return $0.countryCode == topCountry
                }
                return $0.countryCode < $1.countryCode
            } else {
                return $0.name < $1.name
            }
        }
    )
```

File: Example/saltedge-ios/View Controllers/ProvidersViewController.swift [hunk 2]
```diff
@@ -27,25 +27,55 @@ class ProvidersViewController: UIViewController {
...
+                        return true
+                    }
+                }                
+                return $0.countryCode < $1.countryCode
+            } else {
+                return $0.name < $1.name
+            }
+        })
+    }
+
```
Comment (baller784) [inline Example/saltedge-ios/View Controllers/ProvidersViewController.swift:77]: remove extra new lines

File: Example/saltedge-ios/View Controllers/ProvidersViewController.swift [hunk 3]
```diff
@@ -27,25 +27,55 @@ class ProvidersViewController: UIViewController {
...
         tableView.dataSource = self
-        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "ProviderCell")
         searchController.searchResultsUpdater = self
         searchController.hidesNavigationBarDuringPresentation = false
         searchController.dimsBackgroundDuringPresentation = false
         tableView.tableHeaderView = searchController.searchBar
         
+        fetchProviders()
+    }
+
```
Comment (baller784) [inline Example/saltedge-ios/View Controllers/ProvidersViewController.swift:37]: remove extra new lines

File: Example/saltedge-ios/View Controllers/ProvidersViewController.swift [hunk 4]
```diff
@@ -27,25 +27,55 @@ class ProvidersViewController: UIViewController {
...
+    private func fetchProviders() {
         HUD.show(.labeledProgress(title: "Fetching Providers", subtitle: nil))
         SERequestManager.shared.getProviders { [weak self] response in
             switch response {
             case .success(let value):
                 HUD.hide(animated: true)
-                self?.providers = value.data
-                self?.filteredProviders = value.data
-                self?.tableView.reloadData()
+                guard let self = self else { return }
```
Comment (baller784) [inline Example/saltedge-ios/View Controllers/ProvidersViewController.swift:46]: By our internal convention, we add prefix `weak` to weak variables. (e.g. `guard let weakSelf`)

File: Example/saltedge-ios/View Controllers/ProvidersViewController.swift [hunk 5]
```diff
@@ -59,10 +89,18 @@ extension ProvidersViewController: UITableViewDataSource {
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "ProviderCell", for: indexPath)
-        cell.textLabel?.text = filteredProviders[indexPath.row].name
+        
+        if let providerCell = cell as? ProviderCell {
```
Comment (baller784) [inline Example/saltedge-ios/View Controllers/ProvidersViewController.swift:93]: Upper lines can be combined next way:

```swift
guard let cell = tableView.dequeueReusableCell(withIdentifier: "ProviderCell", for: indexPath) as? ProviderCell else { return UITableViewCell() }
```

**Review by baller784 — COMMENTED** [review]
Hello @petalvlad, thank you for your contribution.

**Comment by baller784** [comment]
Hey @petalvlad! There are some merge conflicts. Do you have time to resolve them, otherwise we will finish this merge request for you? Thank you.

**Comment by petalvlad** [comment]
> Hey @petalvlad! There are some merge conflicts. Do you have time to resolve them, otherwise we will finish this merge request for you? Thank you.

Hi @baller784, thank you for your feedback.

Unfortunately I can not make it in a short time, probably could jump back to it in a couple of weeks. 

Feel free to resolve it the way you see it should be.

**Comment by baller784** [comment]
> > Hey @petalvlad! There are some merge conflicts. Do you have time to resolve them, otherwise we will finish this merge request for you? Thank you.
> 
> Hi @baller784, thank you for your feedback.
> 
> Unfortunately I can not make it in a short time, probably could jump back to it in a couple of weeks.
> 
> Feel free to resolve it the way you see it should be.

Ok @petalvlad , we will finish it. Thanks again for your time and contribution!

**Comment by ConstantinKV** [comment]
done in https://github.com/saltedge/saltedge-ios-swift/pull/45

**Pull request closed without merging**